gtsam/hybrid/HybridGaussianFactorGraph.cpp

```diff
@@ -580,4 +580,15 @@ GaussianFactorGraph HybridGaussianFactorGraph::choose(
   return gfg;
 }
 
+/* ************************************************************************ */
+DiscreteFactorGraph HybridGaussianFactorGraph::discreteFactors() const {
+  DiscreteFactorGraph dfg;
+  for (auto &&f : factors_) {
+    auto discreteFactor = std::dynamic_pointer_cast<DiscreteFactor>(f);
+    assert(discreteFactor);
```

Comment:
I don’t understand how this can work. Not all factors are discrete, right?

Reply:
Yup, which is why we have the assert, but I guess the `if(auto discreteFactor = ...)` check is better.